File: 1351.count-negative-numbers-in-a-sorted-matrix.cpp
// @lc app=leetcode id=1351 lang=cpp

#include <bits/stdc++.h>
using namespace std;

// @lc code=start
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int count = 0;
        const int N = grid[0].size();
        int i = N - 1;
        for (const vector<int>& row : grid) {
            while (i >= 0 && row[i] < 0) {
                --i;
            }
            count += N - i - 1;
        }
        return count;
    }
};
// @lc code=end

File: 1351.count-negative-numbers-in-a-sorted-matrix.py
# @lc app=leetcode id=1351 lang=python3

from collections import *
from functools import *
from itertools import *
from operator import *
from typing import *


# @lc code=start
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        count = 0
        N = len(grid[0])
        i = N - 1
        for row in grid:
            while i >= 0 and row[i] < 0:
                i -= 1
            count += N - i - 1
        return count


# @lc code=end
